deploy/role.yaml

```diff
@@ -31,6 +31,7 @@ rules:
   resources:
   - configmaps
   - pods
+  - secrets
```

Comment:
why do we need this?

Reply:
@raffaelespazzoli Currently, the included example CRs do not install due to permissions issues. From what I can see, in order to create a locked resource the service account that's used needs to be able to list/get/watch at a cluster scope as well as create it at the namespace scope (if it's a namespaced resource. 

I changed the example CR to create a **ResourceLocker** that locks a **Secret** because we already have Get/List/Watch for **Secrets** at the cluster scope. The only thing missing was the ability to create secrets at the namespace scope. 

I thought about instead adding Get/List/Watch for **ConfigMaps** at the cluster scope instead (since we already have the ability to create them at the namespace) and then having the demo CR be a **ConfigMap**, but adding cluster scoped permissions seemed like too much. 

Ultimately, the goal is as stated in the original post ("**Goal of this PR**") - allow for an end user to install the included example successfully.

Reply:
I don't like the idea of having to add permissions just to run the examples.
Any other example that would work without adding permissions? Or in alternative would it be ok for you to add a few instructions on prerequisites needed to run the examples?

Reply:
@raffaelespazzoli 

> I don't like the idea of having to add permissions just to run the examples.

Yep, that's totally fair! The only justification I found acceptable was that the service account already had what is effectively "reader" privileges for secrets at the cluster scope so adding "writer" privileges at the namespace_seemed_ less impactful. But you make a good point, it's purely in order to run the example. 

> Any other example that would work without adding permissions?

I tested a few, but couldn't find any that logically functioned out of the box.

> Or in alternative would it be ok for you to add a few instructions on prerequisites needed to run the examples?

We could do this if it's not already there (I do think there's something there already but I would need to double check). Either way, pre-requisite instruction means that the example CR will never build successfully without executing those (which could be missed by users or automation). I'm OK with this trade-off, but it will mark the bugzilla bug I've referenced as WONTFIX. 

Totally your call here! If you want to remove the permission, I'll go ahead and submit that change here, and explain in bugzilla the circumstance and the implications of the design (where the operator has less privilege, relying on the referenced service account to have the privilege needed) to close the bug.

Let me know!